lib/rules/no-multi-assign.js

```diff
@@ -45,16 +45,21 @@ module.exports = {
         const options = context.options[0] || {
             ignoreNonDeclaration: false
         };
-        const targetParent = options.ignoreNonDeclaration ? ["VariableDeclarator"] : ["AssignmentExpression", "VariableDeclarator"];
+        const selectors = [
+            "VariableDeclarator > AssignmentExpression.init",
+            "PropertyDefinition > AssignmentExpression.value"
```

Comment:
I think it makes sense that `"ignoreNonDeclaration": true` does not ignore initializing class fields, but we should probably update the docs.

Current:

> `"ignoreNonDeclaration"`: When set to `true`, the rule allows chains that don't include initializing a variable in a declaration. Default is `false`.

Updated:

> `"ignoreNonDeclaration"`: When set to `true`, the rule allows chains that don't include initializing a variable in a declaration or initializing a class field. Default is `false`.

and it would be good to a class field example in "Examples of incorrect code for the `{ "ignoreNonDeclaration": true }` option".